processor/manager.go

```diff
@@ -56,7 +56,8 @@ func (proc *LifecycleManager) Start() {
 func (proc *LifecycleManager) Stop() {
 	proc.currentCancel()
 	proc.HandleT.Shutdown()
-	proc.waitGroup.Wait()
+	//TODO : Handle the errors cleanly
+	_ = proc.waitGroup.Wait()
```

Comment:
No errors are expected from `waitGroup`, this is why the error is safe to be ignored. Although we should have been implicit about it.

Reply:
The linter pointed it out to handle the error so I added a TODO. Will remove this

Reply:
It looks like something that can be done in a separate PR, in my head when I saw this was happy with the TODO because otherwise we would need `Stop()` to also return an `error` and then have it handled all the way up (e.g. `func (proc *LifecycleManager) Stop() error`).